src/App.js

```diff
@@ -1,86 +1,183 @@
+      })));
+    }
+
+    setAllStatus(!allStatus);
+  };
+
+  const handleEditingTodo = (todoId) => {
+    setTodos(todos.map((todo) => {
+      if (todo.id === todoId) {
+        return {
+          ...todo,
+          isBeingEdited: true,
+        };
+      }
+
+      return {
+        ...todo,
+        isBeingEdited: false,
+      };
+    }));
+  };
+
+  const handleEditedTodo = (todoId, title) => {
+    if (title.trim()) {
+      setTodos(todos.map((todo) => {
+        if (todo.id === todoId) {
+          return {
+            ...todo,
+            isBeingEdited: false,
+            title,
+          };
+        }
+
+        return todo;
+      }));
+
+      return todos;
+    }
+
+    return todos;
+  };
+
+  const handleFilter = (value) => {
+    setFilter(value);
+  };
+
   return (
     <section className="todoapp">
       <header className="header">
         <h1>todos</h1>
 
-        <form>
-          <input
-            type="text"
-            className="new-todo"
-            placeholder="What needs to be done?"
-          />
-        </form>
+        <InputField addNewTodo={addNewTodo} />
       </header>
 
       <section className="main">
-        <input type="checkbox" id="toggle-all" className="toggle-all" />
-        <label htmlFor="toggle-all">Mark all as complete</label>
-
-        <ul className="todo-list">
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>asdfghj</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="completed">
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>qwertyuio</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li className="editing">
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>zxcvbnm</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-
-          <li>
-            <div className="view">
-              <input type="checkbox" className="toggle" />
-              <label>1234567890</label>
-              <button type="button" className="destroy" />
-            </div>
-            <input type="text" className="edit" />
-          </li>
-        </ul>
+        {todos.length === 0 && (<></>)}
+        {(todos.length > 0 && todos.every(todo => todo.completed === true)) && (
+          <>
+            <input
+              checked
+              type="checkbox"
+              id="toggle-all"
+              className="toggle-all"
+              onChange={toggleAll}
+              value={allStatus}
+            />
```

Comment:
use only checked value for checkbox
```suggestion
            <input
              checked={allStatus}
              type="checkbox"
              id="toggle-all"
              className="toggle-all"
              onChange={toggleAll}
            />
```